client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -41,7 +42,7 @@ const siteMigration: Flow = {
 			? [ STEPS.PICK_SITE, STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
 			: [];
 
-		return [ ...baseSteps, ...hostedVariantSteps ];
+		return stepsWithRequiredLogin( [ ...baseSteps, ...hostedVariantSteps ] );
```

Comment:
I think the `stepsWithRequiredLogin` utility slightly damages the simple understanding of which step require auth and which don't. Or the one-to-one connection between a given step and auth. It looks as if it will hide all of these steps behind an auth-wall. 

And since these steps are in the `STEPS` constant, it would be better to fix them centrally and set an example for new steps IMHO.

Reply:
I think a big problem here is that a step on its own can't always dictate whether a user must be logged in or not. As an example, we have onboarding flows that require a user to be logged in for all steps (which is the case here), but some of these steps are used in signup flows where we may want to defer the login/account/user step.

I don't think the `STEPS` constant is the right place to make these determinations for all steps. For some steps, it's definitely required that the user be logged in, regardless of context, so it may make sense for those steps to have the new flag by default.

Reply:
Also, most flows expect all steps to require authentication, so I wanted a clear shorthand for making that happen at a flow level, especially as we roll this out. I'd prefer to ship the initial change to a small set of flows, and then roll it out to more flows once we feel more confident about it in production.

I think we need the shorthand, but I don't have a strong opinion as to what it should look like -- this came up because I didn't want to really mess up the clean array of `STEPS.*` references in `flow.useSteps()`!

Reply:
Oh, yeah! I didn't consider the fact that the same step may or may not require login based on the flow. Let's keep it as is.